cmd/ipfs/main.go

```diff
@@ -595,17 +594,24 @@ func profileIfEnabled() (func(), error) {
 // a running API service. if there is one, it returns a client.
 // otherwise, it returns errApiNotRunning, or another error.
 func getApiClient(repoPath, apiAddrStr string) (cmdsHttp.Client, error) {
-
-	if apiAddrStr == "" {
-		var err error
-		if apiAddrStr, err = fsrepo.APIAddr(repoPath); err != nil {
+	efmt := fmt.Sprintf("failed to parse %[1]s/api file.\n\terror: %%s\n"+
+		"If there is no daemon running, it is safe to delete it.\nYou can do it with:\n"+
+		"\tps aux | grep ipfs # check there is no ipfs daemon\n\trm %[1]s/api\n", repoPath)
+	var addr ma.Multiaddr
+	var err error
+	if len(apiAddrStr) != 0 {
+		addr, err = ma.NewMultiaddr(apiAddrStr)
+		if err != nil {
+			return nil, fmt.Errorf(efmt, err.Error)
+		}
+	} else {
+		addr, err = fsrepo.APIAddr(repoPath)
+		if err != nil {
 			return nil, err
 		}
 	}
-
-	addr, err := ma.NewMultiaddr(apiAddrStr)
-	if err != nil {
-		return nil, err
+	if len(addr.Protocols()) == 0 {
+		return nil, fmt.Errorf(efmt, "multiaddr doesn't provide any protocols")
```

Comment:
While technically correct, i dont think this helps the user understand what to do